WriteActionsTests#testIndex and WriteActionsTests#testBulk failures

On master: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+artifactory/317/console

I couldn't reproduce locally:
```
REPRODUCE WITH: ./gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.security.authz.WriteActionsTests.testIndex" \
  -Dtests.seed=241B472FF1BAA94F \
  -Dtests.security.manager=true \
  -Dtests.locale=ka \
  -Dtests.timezone=Asia/Tomsk \
  -Dcompiler.java=12 \
  -Druntime.java=11

    REPRODUCE WITH: ./gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.security.authz.WriteActionsTests.testBulk" \
  -Dtests.seed=241B472FF1BAA94F \
  -Dtests.security.manager=true \
  -Dtests.locale=ka \
  -Dtests.timezone=Asia/Tomsk \
  -Dcompiler.java=12 \
  -Druntime.java=11
```

There are several errors in the logs that look like this. This looks like sanity checks during cluster tear-down to me, so maybe this is not a test-specific error after all?
```
java.lang.AssertionError: not all translog generations have been released
	at __randomizedtesting.SeedInfo.seed([241B472FF1BAA94F:65CAC4329CD54998]:0)
	at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
	at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$8(InternalTestCluster.java:1276)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:862)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
	at org.elasticsearch.test.InternalTestCluster.assertOpenTranslogReferences(InternalTestCluster.java:1269)
	at org.elasticsearch.test.InternalTestCluster.beforeIndexDeletion(InternalTestCluster.java:1217)
	at org.elasticsearch.test.ESIntegTestCase.beforeIndexDeletion(ESIntegTestCase.java:598)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:568)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2170)
	at jdk.internal.reflect.GeneratedMethodAccessor47.invoke(Unknown Source)
	at 
```

Got another failure in another class but with 
```
java.lang.AssertionError: not all translog generations have been released
	at __randomizedtesting.SeedInfo.seed([A800D548D12BE107:5A9E1E500F3502CA]:0)
	at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
	at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$8(InternalTestCluster.java:1276)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:862)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
	at org.elasticsearch.test.InternalTestCluster.assertOpenTranslogReferences(InternalTestCluster.java:1269)
	at org.elasticsearch.test.InternalTestCluster.beforeIndexDeletion(InternalTestCluster.java:1217)
	at org.elasticsearch.test.ESIntegTestCase.beforeIndexDeletion(ESIntegTestCase.java:598)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:568)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2170)
```
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=ubuntu-14.04&&immutable/325/console

```
./gradlew :x-pack:plugin:security:test --tests "org.elasticsearch.xpack.security.authz.IndexAliasesTests.testCreateIndexAndAliasesCreateAndAliasesPermission" \
  -Dtests.seed=A800D548D12BE107 \
  -Dtests.security.manager=true \
  -Dtests.locale=it \
  -Dtests.timezone=Europe/Malta \
  -Dcompiler.java=12 \
  -Druntime.java=11
```
I start thinking this is a more general thing.

There's another recent failure here: https://scans.gradle.com/s/ihktsg6wdryoi

While the console output on the build scan does not contain the full assertion failure (it's just the top-level message, not the suppressed exceptions, not sure why, perhaps @atorok or @mark-vieira know?)  but the e-mail fortunately has the full stack trace, repeated here:

```
java.lang.AssertionError: not all translog generations have been released
	at __randomizedtesting.SeedInfo.seed([507EA81B7B24480C:9CAFEF0054DBEA5]:0)
	at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
	at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:862)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
	at org.elasticsearch.test.InternalTestCluster.assertOpenTranslogReferences(InternalTestCluster.java:1338)
	at org.elasticsearch.test.InternalTestCluster.beforeIndexDeletion(InternalTestCluster.java:1286)
	at org.elasticsearch.test.ESIntegTestCase.beforeIndexDeletion(ESIntegTestCase.java:622)
	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:592)
	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2194)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:835)
	Suppressed: java.lang.RuntimeException
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
		at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
		at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
		at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
		at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
		at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
		at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
		at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
		at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
		at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
		at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
		at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
		at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
		at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
		at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
		at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
		at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
		at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
		at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
		at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
		at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
		at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
		at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
		at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
		at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
		at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
		at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
		at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
		at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		... 1 more
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
		Suppressed: java.lang.RuntimeException
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
			at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
			at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
			at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
			at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
			at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
			at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
			at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
			at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
			at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
			... 1 more
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
		Suppressed: java.lang.RuntimeException
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
			at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
			at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
			at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
			at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
			at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
			at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
			at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
			at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
			at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
			... 1 more
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
		Suppressed: java.lang.RuntimeException
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
			at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
			at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
			at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
			at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
			at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
			at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
			at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
			at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
			at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
			... 1 more
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
	[CIRCULAR REFERENCE:java.lang.RuntimeException]
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
		Suppressed: java.lang.RuntimeException
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
			at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
			at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
			at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
			at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
			at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
			at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
			at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
			at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
			at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
			... 1 more
	Suppressed: java.lang.AssertionError: not all translog generations have been released
		at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertNoOpenTranslogRefs(TranslogDeletionPolicy.java:39)
		at org.elasticsearch.test.InternalTestCluster.lambda$assertOpenTranslogReferences$9(InternalTestCluster.java:1345)
		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:850)
		... 42 more
		Suppressed: java.lang.RuntimeException
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.assertAddTranslogRef(TranslogDeletionPolicy.java:120)
			at org.elasticsearch.index.translog.TranslogDeletionPolicy.acquireTranslogGen(TranslogDeletionPolicy.java:110)
			at org.elasticsearch.index.translog.Translog.acquireTranslogGenFromDeletionPolicy(Translog.java:709)
			at org.elasticsearch.index.translog.Translog.acquireRetentionLock(Translog.java:704)
			at org.elasticsearch.index.engine.InternalEngine.acquireRetentionLock(InternalEngine.java:2594)
			at org.elasticsearch.index.shard.IndexShard.acquireRetentionLock(IndexShard.java:1735)
			at org.elasticsearch.indices.recovery.RecoverySourceHandler.recoverToTarget(RecoverySourceHandler.java:158)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.recover(PeerRecoverySourceService.java:106)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService.access$000(PeerRecoverySourceService.java:54)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:112)
			at org.elasticsearch.indices.recovery.PeerRecoverySourceService$StartRecoveryTransportRequestHandler.messageReceived(PeerRecoverySourceService.java:109)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler$1.doRun(SecurityServerTransportInterceptor.java:250)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:192)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.lambda$messageReceived$0(SecurityServerTransportInterceptor.java:299)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorizeSystemUser(AuthorizationService.java:375)
			at org.elasticsearch.xpack.security.authz.AuthorizationService.authorize(AuthorizationService.java:183)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.lambda$inbound$1(ServerTransportFilter.java:130)
			at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$authenticateAsync$2(AuthenticationService.java:246)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lambda$lookForExistingAuthentication$6(AuthenticationService.java:306)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.lookForExistingAuthentication(AuthenticationService.java:317)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.authenticateAsync(AuthenticationService.java:244)
			at org.elasticsearch.xpack.security.authc.AuthenticationService$Authenticator.access$000(AuthenticationService.java:196)
			at org.elasticsearch.xpack.security.authc.AuthenticationService.authenticate(AuthenticationService.java:139)
			at org.elasticsearch.xpack.security.transport.ServerTransportFilter$NodeProfile.inbound(ServerTransportFilter.java:121)
			at org.elasticsearch.xpack.security.transport.SecurityServerTransportInterceptor$ProfileSecuredRequestHandler.messageReceived(SecurityServerTransportInterceptor.java:306)
			at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:63)
			at org.elasticsearch.transport.InboundHandler$RequestHandler.doRun(InboundHandler.java:267)
			at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:758)
			at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
			at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
			at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
			... 1 more
```

It looks like the retention lock from peer recovery was not freed, which could have multiple causes: 1) peer recovery did not complete yet (the test does not wait for green), or 2) we effectively leaked the retention lock during peer recovery. What's surprising though is that this would only happen in these particular tests, which is why I'm leaning towards the former. @dnhatn Should we do an `ensureGreen` at the end of the test and hope for the best?

--- a/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java
@@ -1202,7 +1202,7 @@ public final class InternalTestCluster extends TestCluster {
                     }
                 }
             }
-        });
+        }, 60, TimeUnit.SECONDS);
     }
 
     private void assertNoSnapshottedIndexCommit() throws Exception {

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/WriteActionsTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/WriteActionsTests.java
@@ -53,6 +53,7 @@ public class WriteActionsTests extends SecurityIntegTestCase {
         //the missing index gets automatically created (user has permissions for that), but indexing fails due to missing authorization
         assertThrowsAuthorizationExceptionDefaultUsers(client().prepareIndex("missing", "type", "id").setSource("field", "value")::get,
                 BulkAction.NAME + "[s]");
+        ensureGreen();
     }
 
     public void testDelete() {
@@ -63,6 +64,7 @@ public class WriteActionsTests extends SecurityIntegTestCase {
         assertThrowsAuthorizationExceptionDefaultUsers(client().prepareDelete("index1", "type", "id")::get, BulkAction.NAME + "[s]");
 
         expectThrows(IndexNotFoundException.class, () -> client().prepareDelete("test4", "type", "id").get());
+        ensureGreen();
     }
 
     public void testUpdate() {
@@ -79,6 +81,7 @@ public class WriteActionsTests extends SecurityIntegTestCase {
 
         assertThrowsAuthorizationExceptionDefaultUsers(client().prepareUpdate("missing", "type", "id")
                         .setDoc(Requests.INDEX_CONTENT_TYPE, "field2", "value2")::get, UpdateAction.NAME);
+        ensureGreen();
     }
 
     public void testBulk() {
@@ -160,5 +163,6 @@ public class WriteActionsTests extends SecurityIntegTestCase {
         assertAuthorizationExceptionDefaultUsers(bulkResponse.getItems()[12].getFailure().getCause(), BulkAction.NAME + "[s]");
         assertThat(bulkResponse.getItems()[12].getFailure().getCause().getMessage(),
                 containsString("[indices:data/write/bulk[s]] is unauthorized"));
+        ensureGreen();
     }
 }

